src/Serilog/Settings/KeyValuePairs/KeyValuePairSettings.cs

```diff
@@ -185,6 +252,36 @@ internal static IEnumerable<Assembly> LoadConfigurationAssemblies(Dictionary<str
             return configurationAssemblies.Distinct();
         }
 
+        internal static bool TryInstantiate(Type type, string constructorParamAsString, out object createdInstance)
```

Comment:
I'm wondering if I did not over-engineer that part ... Seeing how the "callable directives" relied on reflection to find the proper method to call, I did the same to try and find a constructor for the type we are trying to build .... but considering the only type we need to build is always `LoggingLevelSwitch`, that might be overkill ... 
If you think this should be simplified in order not to rely on reflection, that's fine, I'll do it :p

Reply:
I'm just doing a quick first-pass over to try to provide some feedback, so haven't give it a huge amount of thought yet, but my instinct says just building something explicit to support `LevelSwitch` would be reasonable right now. 

We're likely to address general object construction through the requirement that sinks be able to receive complex parameters to their configuration methods - `ColumnOptions` for the MSSQL sink, for example. My bet is we'll revisit general construction rules then.

Reply:
ok, go for YAGNI then :)